File: Assets/Scripts/Bullet.js
#pragma strict

public var thrust : float = 10;

function Start () {

	var player : GameObject = gameObject.Find("Player");
	var playerMovement : PlayerMovement = player.GetComponent(PlayerMovement);
	
	var radRotation = playerMovement.radRotation;
	var playerVelX = player.rigidbody2D.velocity.x;
	var playerVelY = player.rigidbody2D.velocity.y;
	
	var vx = -(Mathf.Sin(radRotation) * thrust) + playerVelX;
	var vy = (Mathf.Cos(radRotation) * thrust) + playerVelY;
	
	rigidbody2D.velocity = (Vector2(vx, vy));

}

function OnBecameInvisible () {
	Destroy(gameObject);
}

File: Assets/Scripts/Bullet.cs
using UnityEngine;
using System.Collections;

public class Bullet : MonoBehaviour {

	public float thrust = 10;
	public float damageDealt = 1;
	private GameObject player;
	private PlayerMovement playerMovement;
	private TargetEnemies te;

	// Use this for initialization
	void Start () {
		player = GameObject.Find ("Player");
		playerMovement = player.GetComponent<PlayerMovement>();
		te = player.GetComponent<TargetEnemies> ();

		Vector2 dir = (te.selectedTarget.transform.position - transform.position).normalized * thrust;
		rigidbody2D.velocity = dir;
	}
	
	// Update is called once per frame
	void OnBecameInvisible () {
		Destroy(gameObject);
	}
}
